### PR TITLE
removed s3 and use api calls

### PR DESCRIPTION
**Pull request by gferraro**
requires api PR https://github.com/TheCacophonyProject/cacophony-web/pull/14

### CODE REVIEW

File: processing/api.py [hunk 1]
```diff
@@ -122,3 +127,65 @@ def add_track_tag(self, recording, track_id, prediction, data=""):
...
+    def download_file(self, token, filename):
+        r = requests.get(
+            urljoin(self.api_url, "/api/v1/signedUrl"),
+            params={"jwt": token},
+            stream=True,
+        )
+        r.raise_for_status()
+        return iter_to_file(filename, r.iter_content(chunk_size=4096))
+
+    def upload_file(self, token, filename):
```
Comment (hardiesoft) [inline processing/api.py:140]: Is this function supposed to be here?

File: processing/api.py [hunk 2]
```diff
@@ -122,3 +127,65 @@ def add_track_tag(self, recording, track_id, prediction, data=""):
...
+            urljoin(self.api_url, "/signedUrl"),
+            params={"jwt": token},
+            stream=True,
+        )
+        r.raise_for_status()
+        return iter_to_file(filename, r.iter_content(chunk_size=4096))
+
+    def upload_file(self, filename):
+        url = self.file_url + "/processed"
+        data = {"fileHash": sha_hash(filename)}
```
Comment (hardiesoft) [inline processing/api.py:151]: Could be wrong, but looks like you're hashing the filename here rather than the file contents?

**Review by hardiesoft — CHANGES_REQUESTED** [review]
Looks good, but would be good to also remove unused deps from requirements.txt - I think there are a couple things there now, including boto, maybe numba too?

**Comment by gferraro** [comment]
> Looks good, but would be good to also remove unused deps from requirements.txt - I think there are a couple things there now, including boto, maybe numba too?

good call, numba is required by librosa for some reason pip doesn't install both

**Review by hardiesoft — APPROVED** [review]

**Pull request merged**